app/src/main/java/nu/yona/app/api/manager/impl/ActivityCategoryManagerImpl.java

```diff
@@ -52,33 +53,13 @@ public ActivityCategoryManagerImpl(Context context) {
      * @param listener
      */
     @Override
-    public void getActivityCategoriesById(final DataLoadListener listener) {
-        try {
-            User user = YonaApplication.getEventChangeManager().getDataState().getUser();
-            if (user != null && user.getLinks() != null && user.getLinks().getYonaAppActivity() != null && !TextUtils.isEmpty(user.getLinks().getYonaAppActivity().getHref())) {
-                activityCategoriesNetwork.getActivityCategories(authenticateDao.getUser().getLinks().getYonaAppActivity().getHref(), new DataLoadListener() {
-                    @Override
-                    public void onDataLoad(Object result) {
-                        updateActivityCategories(result, null);
-                        if (listener != null) {
-                            listener.onDataLoad(result);
-                        }
-                    }
-
-                    @Override
-                    public void onError(Object errorMessage) {
-                        if (listener != null) {
-                            listener.onError(errorMessage);
-                        }
-                    }
-                });
-            } else {
-                if (listener != null) {
-                    listener.onError(new ErrorMessage(mContext.getString(R.string.urlnotfound)));
-                }
-            }
-        } catch (Exception e) {
-            AppUtils.throwException(ActivityCategoryManagerImpl.class.getSimpleName(), e, Thread.currentThread(), listener);
+    public void getActivityCategoriesById(DataLoadListener listener) {
+        User user = YonaApplication.getEventChangeManager().getDataState().getUser();
+        if (user != null && user.getLinks() != null && user.getLinks().getYonaAppActivity() != null && !TextUtils.isEmpty(user.getLinks().getYonaAppActivity().getHref())) {
```

Comment:
This is a terrible statement too. Can ``user` reasonably be ``null`` here? Is it reasonable to expect that the user doesn't have an app activity URL?

Reply:
Dont know if the guys had this checks after getting some issue or they were over cautious.

Reply:
We shouldn't do that, as it's a lot of clutter.